### PR TITLE
Support dropping default for a column

### PR DESCRIPTION
**Pull request by ryanslade**
We no allow setting the value of `default` to `null` when altering a column. In practice this will cause us to drop the default. It is also possible to set `default` the *value* `NULL` which has a similar effect but is recorded slightly differently in the PG catalogue. (Details [here](https://github.com/xataio/pgroll/issues/450#issuecomment-2490680272))

Closes https://github.com/xataio/pgroll/issues/450

### CODE REVIEW

File: pkg/migrations/op_set_default_test.go [hunk 1]
```diff
@@ -248,7 +250,109 @@ func TestSetDefault(t *testing.T) {
 						&migrations.OpAlterColumn{
 							Table:   "users",
 							Column:  "name",
-							Default: ptr("NULL"),
+							Default: nullable.NewNullableWithValue("NULL"),
```
Comment (ryanslade) [inline pkg/migrations/op_set_default_test.go:253]: This diff is very confusing. What I actually did was duplicate the test where we set the value to `NULL` with another example where we set it to `null`.

`Default: nullable.NewNullableWithValue("NULL")` vs `Default: nullable.NewNullNullable[string](),`

File: docs/README.md [hunk 2]
```diff
@@ -862,7 +862,7 @@ A change default operation changes the default value of a column.
   "alter_column": {
     "table": "table name",
     "column": "column name",
-    "default": "new default value",
+    "default": "new default value (or null)",
```
Comment (andrew-farries) [inline docs/README.md:865]:
```suggestion
    "default": "new default value" | null,
```

File: examples/46_alter_column_drop_default.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "46_alter_column_drop_default",
+  "operations": [
+    {
+      "alter_column": {
+        "table": "tickets",
+        "column": "ticket_type",
+        "default": null,
+        "up": "",
+        "down": ""
```
Comment (andrew-farries) [inline examples/46_alter_column_drop_default.json:10]: The data migration in both directions can be a straight copy without modification:

```suggestion
        "up": "ticket_type",
        "down": "ticket_type"
```

**Review by andrew-farries — APPROVED** [review]
This looks good, just a couple of small suggestions.

**Pull request merged**